File: src/main/java/com/github/cndjp/godfather/event/connpass/ConnpassParticipant.java
package com.github.cndjp.godfather.event.connpass;

import java.net.URL;
import com.github.cndjp.godfather.event.Participant;
import com.github.cndjp.godfather.event.ParticipationStatus;

public class ConnpassParticipant implements Participant {

    private final String id;

    private final String displayName;

    private final URL imageSource;

    private final ParticipationStatus participationStatus;

    public ConnpassParticipant(final String id, final String displayName,
                               final URL imageSource, final ParticipationStatus participationStatus) {
        this.id = id;
        this.displayName = displayName;
        this.imageSource = imageSource;
        this.participationStatus = participationStatus;
    }

    @Override
    public String getId() {
        return this.id;
    }

    @Override
    public String getDisplayName() {
        return this.displayName;
    }

    @Override
    public URL getImageSource() {
        return this.imageSource;
    }

    @Override
    public ParticipationStatus getParticipationStatus() {
        return this.participationStatus;
    }

    @Override
    public String toString() {
        return "ConnpassParticipant{" +
                "id='" + id + '\'' +
                ", displayName='" + displayName + '\'' +
                ", imageSource=" + imageSource +
                ", participationStatus=" + participationStatus +
                '}';
    }

}

File: src/main/scala/com/github/cndjp/godfather/domain/participant/ConnpassParticipant.scala
package com.github.cndjp.godfather.domain.participant

import java.net.URL
import java.util.UUID
import org.jsoup.nodes.{Document, Element}
import com.github.cndjp.godfather.utils.GodfatherConstants._

// connpassイベントに参加する人を表すドメインクラス
case class ConnpassParticipant(id: String, name: String, imageURL: URL) extends Participant

object ConnpassParticipant {

  def apply(displayName: String, userDoc: Document): ConnpassParticipant = {
    val images = userDoc.select("div[id=side_area] div[class=mb_20 text_center] a.image_link")
    val imageSource =
      if (!images.isEmpty)
        images
          .toArray(Array[Element]())
          .find(_.attr("href")
            .contains("/user/"))
          .map(image => new URL(image.attr("href")))
          .getOrElse(IMAGE_SOURCE_DEFAULT)
      else IMAGE_SOURCE_DEFAULT
    ConnpassParticipant(
      UUID.randomUUID().toString,
      displayName,
      imageSource,
    )
  }
}
